torch/nested/_internal/ops.py

```diff
@@ -172,19 +192,226 @@ def tensor_attr_unsupported_getter(func, *args, **kwargs):
 
 @register_jagged_func(torch.ops.aten.linear.default, "input: jt, weight: t, bias: t?")
 def linear_default(func, *args, **kwargs):
-    values = torch.mm(args[0].values(), args[1])
-    if len(args) == 3:
-        values += args[2]
-    return NestedTensor(values, **extract_kwargs(args[0]))
+    _, new_kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+
+    inp = new_kwargs.pop("input")
+    weight = new_kwargs["weight"]
+    bias = new_kwargs["bias"]
+
+    new_values = torch.mm(inp._values, weight)
+    if bias is not None:
+        new_values += bias
+    return NestedTensor(new_values, inp._offsets)
```

Comment:
Unfortunately, for the symbolic singleton symint to be propagated, we should also need to thread the `inp._ragged_size` around. For now I'm using `extract_kwargs` as a utility to propagate all necessary metadata.

```suggestion
    return NestedTensor(new_values, **extract_kwargs(inp))
```

Reply:
Oh yeah this is probably rebase shenanigans, thanks for doing the rebase :)